File: codegen_sources/test_generation/evosuite_tests_translators/tests/resources/expected_python_translations/integer_array_test.py
import numpy as np
import math
from math import *
import collections
from collections import *
import heapq
import itertools
import random
import sys
import unittest


# TOFILL


class PERMUTE_TWO_ARRAYS_SUM_EVERY_PAIR_GREATER_EQUAL_K(unittest.TestCase):
    def test0(self):
        integerArray0 = [None] * 2
        int0 = -1
        integer0 = -1
        assert (-1) == int(integer0)
        assert integer0 == (int0)
        assert integer0 is not None

        integerArray0[0] = integer0
        integer1 = 1
        assert 1 == int(integer1)
        assert not (integer1 == (int0))
        assert not (integer1 == (integer0))
        assert integer1 is not None

        integerArray0[1] = integer1
        intArray0 = [0] * 3
        intArray0[2] = int0
        boolean0 = f_filled(integerArray0, intArray0, 1, 0)
        assert boolean0
        assert [(-1), 0, 0] == intArray0
        assert 2 == len(integerArray0)
        assert 3 == len(intArray0)

    def test1(self):
        integerArray0 = [None] * 2
        int0 = -1
        integer0 = -1
        assert (-1) == int(integer0)
        assert integer0 == (int0)
        assert integer0 is not None

        integerArray0[0] = integer0
        int1 = 1
        integerArray0[1] = integer0
        intArray0 = [0] * 3
        intArray0[2] = int0
        boolean0 = f_filled(integerArray0, intArray0, int1, (-50146))
        assert boolean0
        assert [(-1), 0, 0] == intArray0
        assert not (int1 == int0)
        assert 2 == len(integerArray0)
        assert 3 == len(intArray0)

    def test2(self):
        integerArray0 = [None] * 2
        integer0 = -1
        assert (-1) == int(integer0)
        assert integer0 is not None

        integerArray0[0] = integer0
        integerArray0[1] = integer0
        intArray0 = [0] * 3
        boolean0 = f_filled(integerArray0, intArray0, (-54229), 1)
        assert boolean0
        assert [0, 0, 0] == intArray0
        assert 2 == len(integerArray0)
        assert 3 == len(intArray0)

    def test3(self):
        integerArray0 = [None] * 2
        integer0 = -1
        assert (-1) == int(integer0)
        assert integer0 is not None

        integerArray0[0] = integer0
        int0 = 1
        integerArray0[1] = integerArray0[0]
        intArray0 = [0] * 3
        boolean0 = f_filled(integerArray0, intArray0, 1, int0)
        assert not (boolean0)
        assert [0, 0, 0] == intArray0
        assert 2 == len(integerArray0)
        assert 3 == len(intArray0)


if __name__ == "__main__":
    unittest.main()

File: codegen_sources/test_generation/evosuite_tests_translators/tests/resources/expected_cpp_translations/integer_array_test.cpp
#include <iostream>
#include <cstdlib>
#include <string>
#include <vector>
#include <fstream>
#include <iomanip>
#include <bits/stdc++.h>
#include "gtest/gtest.h"
#include "gmock/gmock.h"

using namespace std;



//TOFILL





  TEST(EvoSuiteTest, test0){
      int* integerArray0 = new int[2];
      int int0 = (-1);
      int integer0 = ((-1));
      ASSERT_EQ ((-1),  (int)integer0);
      ASSERT_TRUE (integer0 == (int0));

      integerArray0[0] = integer0;
      int integer1 = (1);
      ASSERT_EQ (1,  (int)integer1);
      ASSERT_FALSE (integer1 == (int0));
      ASSERT_FALSE (integer1 == (integer0));

      integerArray0[1] = integer1;
      int* intArray0 = new int[3];
      intArray0[2] = int0;
      bool bool0 = f_filled(integerArray0, intArray0, 1, 0);
      ASSERT_TRUE (bool0);
      ASSERT_THAT(new int[] {(-1), 0, 0}, ::testing::ContainerEq( intArray0));
      ASSERT_EQ (2,  integerArray0.length);
      ASSERT_EQ (3,  intArray0.length);
  }

  TEST(EvoSuiteTest, test1){
      int* integerArray0 = new int[2];
      int int0 = (-1);
      int integer0 = ((-1));
      ASSERT_EQ ((-1),  (int)integer0);
      ASSERT_TRUE (integer0 == (int0));

      integerArray0[0] = integer0;
      int int1 = 1;
      integerArray0[1] = integer0;
      int* intArray0 = new int[3];
      intArray0[2] = int0;
      bool bool0 = f_filled(integerArray0, intArray0, int1, (-50146));
      ASSERT_TRUE (bool0);
      ASSERT_THAT(new int[] {(-1), 0, 0}, ::testing::ContainerEq( intArray0));
      ASSERT_FALSE (int1 == int0);
      ASSERT_EQ (2,  integerArray0.length);
      ASSERT_EQ (3,  intArray0.length);
  }

  TEST(EvoSuiteTest, test2){
      int* integerArray0 = new int[2];
      int integer0 = ((-1));
      ASSERT_EQ ((-1),  (int)integer0);

      integerArray0[0] = integer0;
      integerArray0[1] = integer0;
      int* intArray0 = new int[3];
      bool bool0 = f_filled(integerArray0, intArray0, (-54229), 1);
      ASSERT_TRUE (bool0);
      ASSERT_THAT(new int[] {0, 0, 0}, ::testing::ContainerEq( intArray0));
      ASSERT_EQ (2,  integerArray0.length);
      ASSERT_EQ (3,  intArray0.length);
  }

  TEST(EvoSuiteTest, test3){
      int* integerArray0 = new int[2];
      int integer0 = ((-1));
      ASSERT_EQ ((-1),  (int)integer0);

      integerArray0[0] = integer0;
      int int0 = 1;
      integerArray0[1] = integerArray0[0];
      int* intArray0 = new int[3];
      bool bool0 = f_filled(integerArray0, intArray0, 1, int0);
      ASSERT_FALSE (bool0);
      ASSERT_THAT(new int[] {0, 0, 0}, ::testing::ContainerEq( intArray0));
      ASSERT_EQ (2,  integerArray0.length);
      ASSERT_EQ (3,  intArray0.length);
  }



int main(int argc, char **argv) {
  ::testing::InitGoogleTest(&argc, argv);
  return RUN_ALL_TESTS();
}